codazen/webDiplomacy
# feature/fbfw-110-138-coastal-inland-builds

### Summary
This PR adds builds to coastal and inland territories. Depending on the territory type, you will be able to build either an army or a fleet. This PR has not implemented building units in a map oddity like St. Petersburg. 

Rules: 
1. You cannot build on a territory that already has a unit
2. You cannot build on supply center territories that are not your home (starting) territories. For example, Austria can only build in Vienna, Trieste, and Budapest. 
3. You cannot build more units than you have supply centers. For example, if you have 5 supply centers and 4 units, you can only build ONE army or ONE fleet (depending on the territory) even if you have all your home supply center territories available. 
4. You CANNOT build on home territories that do not have a supply center. 

### Videos
This video shows the user as the country of Austria building on Trieste and Budapest. The user has 6 supply centers and only 3 units, but he is only able to build 2 units because there is already a unit stationed in Vienna. 
https://drive.google.com/file/d/1eiTHVR9vp7q4vQibP8n3r9Tcaoonoe4l/view?usp=sharing

This next video shows the user as the country of Germany having the option to build on Berlin, Kiel or Munich. However, this user has 4 supply centers and 3 units, which he means he can only build on one of those three home territories, despite all of them being available. 
https://drive.google.com/file/d/144Uq3T7JyNMpJfwLzsbkZYeT1xv44KVT/view?usp=sharing

### Testing Plan
1. Start up your local environment and go to an existing game or start a new game
2. Proceed to builds phase
3. Click on a valid territory to build on that territory. Depending on the territory you may be able to build an army or both a fleet and an army. 
4. Save your build orders and verify that they carried to the legacy app. 
5. Also ensure that when loading the beta app, existing build orders are shown. 

### Links to Relevant Tasks
https://codazen.atlassian.net/browse/FBFW-110
https://codazen.atlassian.net/browse/FBFW-138
